stdlib/public/Concurrency/TaskLocal.cpp

```diff
@@ -223,7 +223,7 @@ void TaskLocal::Item::copyTo(AsyncTask *target) {
   assert(target && "TaskLocal item attempt to copy to null target task!");
 
   auto item = Item::createLink(target, this->key, this->valueType);
-  valueType->vw_initializeWithTake(item->getStoragePtr(), this->getStoragePtr());
+  valueType->vw_initializeWithCopy(item->getStoragePtr(), this->getStoragePtr());
```

Comment:
This is in the "copy to target task" which we only do for unstructured tasks.

I'm not 100% sure this is the right way to fix but I _think_ so?

Reply:
Right, you need the "copy" here because the source needs to remain valid.